Add contribution instructions to search-collector README. (#129)
open-cluster-management/search-collector: readme.md

RUNTIME_MODE    | no       | production             | Running mode (development or production)

### Other Configuration Options
- Environment variables can also be set in the `./config.json` for development. If both provide a value for a specific property, the environment variable overrides the file. You can define your own `config.json` file and pass it to the application with the following command: `-c <config_file>`
- The application can take any flags for [glog](https://github.com/golang/glog), which passes them straight into glog. The glog flag `--logtostderr` is set to true by default.

### Contribution
When you make contributions to this project, fork the project then merge your changes by creating a pull request from your personal fork to the main project:

1. Fork the `search-collector` repository.
2. Clone your fork.
3. Set upstream origin with the following command: `git remote add upstream git@github.com:open-cluster-management/search-collector.git`
4. Make your edits. Test locally with `make test` before creating a pull request to ensure smooth merging :smile:
5. Push your new commits to your personal fork with the following command: `git push origin`

> 6. Create a pull request from your personal fork again the upstream `search-collector` main branch

Reviewer: @mqhackett should it be from your personal fork **against** the upstream ...?

### Other Configuration Options
- Environment variables can also be set in ./config.json for development. If both provide a value for a specific property, the environment variable will override the file. You can define your own config json file and pass it to the application with `-c <config_file>`

> - The application can take any flags for [glog](https://github.com/golang/glog), it will pass them straight into glog. The glog flag `--logtostderr` is set to true by default.

Author: Update document title capitalization
changed `search collector` to `search-collector` (added dash)
updated data model section to include link
Updated Usage and configuration capitaliztion

I left the prerequisites section as is. The tips are listed as steps that sometimes are required to fetch the dependencies, step one of the steps.

### Contribution

> When making contributions to the search collector project, fork the project then merge your chages by creating a pull request from your personal fork to the main project.

Reviewer 2: 1. Keep generic to make it easy to copy/paste and maintain across our repos.
2. Fix typo.

```suggestion
When making contributions to this project, fork the project then merge your changes by creating a pull request from your personal fork to the main project.
```

Reviewer: I suggest that the statement reads: 

When you make contributions to this project, fork the project  then merge your changes by creating a pull request from your personal fork to the main project:

Author: 👍

1. Fork Repository
2. Clone your fork
3. Set upstream origin `set upstream origin git@github.com:open-cluster-management/search-collector.git`

> 4. Make your edits. Please test them locally before creating a pull request to ensure smooth merging :smile:

Reviewer 2: Add a pointer to how to test locally.
```suggestion
4. Make your edits. Test locally with `make test` before creating a pull request to ensure smooth merging :smile:
```

Reviewer: Be sure to have punctuation to end the statements. 

1. Fork the `search-collector` repository.
2. Clone your fork. 
3. Set upstream origin with the following command:

Author: I added punctuation. 
I changed the command to set origin to `git remote add upstream git@github.com:open-cluster-management/search-collector.git`

> 5. Push your new commits to your personal fork `git push origin`

Reviewer: Push your new commits to your personal fork with the following command:

> 6. Create a pull request from your personal fork again the upstream search-collector main branch

Reviewer: please make search-collector monospaced: `search-collector`

Author: 👍
